output-gazetteer.cpp

```diff
@@ -742,7 +740,7 @@ int output_gazetteer_t::process_relation(osmid_t id, struct member *members,
         return 0;
 
     /* get the boundary path (ways) */
-    osmid_t *xid2 = new osmid_t[member_count+1];
+    std::vector<osmid_t> xid2(member_count+1);
```

Comment:
`xid2[count] = members[i].id;` below could become xid2.push_back(members[i].id), and the `count == 0` check could then use xid2.size()